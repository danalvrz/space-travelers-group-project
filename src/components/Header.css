a.selected {
    text-decoration: none;
}
a.selected:active {
    text-decoration: underline;
}