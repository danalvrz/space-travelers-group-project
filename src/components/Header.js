import React from 'react';
import { NavLink } from 'react-router-dom';
import logo from '../planet.png';

function Header() {
  return (
    <header className="menuContainer flex justify-evenly container mt-5 mb-5 pb-6 border-b mx-auto">
      <div className="flex container mr-20">
        <span><img src={logo} alt="planet" width={60} /></span>
        <h1 className="font-bold text-4xl px-5 mt-1">Space Travelers&lsquo; Hub</h1>
      </div>
      <nav className="menuList mt-5 container ml-20 pl-20">
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'text-blue-500 font-bold underline mx-5' : 'text-black font-thin mx-5')}
        >
          Rockets

        </NavLink>
        <NavLink
          to="/missions"
          className={({ isActive }) => (isActive ? 'text-blue-500 font-bold underline mx-5' : 'text-black font-thin mx-5')}
        >
          Missions

        </NavLink>
        <NavLink
          to="/myProfile"
          className={({ isActive }) => (isActive ? 'text-blue-500 font-bold underline mx-5' : 'text-black font-thin mx-5')}
        >
          My Profile

        </NavLink>
      </nav>
    </header>
  );
}

export default Header;
