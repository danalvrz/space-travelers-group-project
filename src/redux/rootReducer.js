import { combineReducers } from 'redux';
import missions from './missions/missions';
import rockets from './rockets/rockets';

const rootReducer = combineReducers({
  missions,
  rockets,
});

export default rootReducer;
