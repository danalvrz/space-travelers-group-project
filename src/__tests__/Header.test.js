import React from 'react';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createStore, combineReducers } from 'redux';
import Header from '../components/Header';

const mockStore = [];
const mockMissionsReducer = () => [{
  mission_id: 0, mission_name: 'test', mission_description: 'test', member: true,
}];
const mockRocketsReducer = () => mockStore;
const mockReducer = () => combineReducers({
  mockMissionsReducer,
  mockRocketsReducer,
});
const store = createStore(mockReducer);

describe('Header section', () => {
  it('renders Header correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
