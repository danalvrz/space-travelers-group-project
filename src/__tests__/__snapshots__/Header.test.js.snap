// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header section renders Header correctly 1`] = `
<header
  className="menuContainer flex justify-evenly container mt-5 mb-5 pb-6 border-b ml-20"
>
  <div
    className="flex container mr-20"
  >
    <span>
      <img
        alt="planet"
        src="planet.png"
        width={60}
      />
    </span>
    <h1
      className="font-bold text-4xl px-5 mt-1"
    >
      Space Travelersâ€˜ Hub
    </h1>
  </div>
  <nav
    className="menuList mt-5 container ml-20 pl-20"
  >
    <a
      className="text-black font-thin mx-5"
      href="#/rockets"
      onClick={[Function]}
    >
      Rockets
    </a>
    <a
      className="text-black font-thin mx-5"
      href="#/missions"
      onClick={[Function]}
    >
      Missions
    </a>
    <a
      className="text-black font-thin mx-5"
      href="#/myProfile"
      onClick={[Function]}
    >
      My Profile
    </a>
  </nav>
</header>
`;
